flink-connector-examples/src/main/java/io/pravega/example/flink/watermark/EventTimeAverage.java

```diff
@@ -104,14 +130,16 @@ public Double getResult(Tuple2<Integer, Double> acc) {
     }
 
     private static class WindowProcess extends ProcessWindowFunction<Double,
-            Tuple4<Integer, Long, Long, Double>, Tuple, TimeWindow> {
+            SensorData, Tuple, TimeWindow> {
 
         @Override
         public void process(Tuple key, Context context, Iterable<Double> iterable,
-                            Collector<Tuple4<Integer, Long, Long, Double>> collector) throws Exception {
+                            Collector<SensorData> collector) throws Exception {
             Double avg = iterable.iterator().next();
             int sensorId = key.getField(0);
-            collector.collect(new Tuple4<>(sensorId, context.window().getStart(), context.window().getEnd(), avg));
+            // Set the event timestamp to the middle timestamp of the window
```

Comment:
I believe that any records produced by the window function have a record timestamp equal to the window's end time.   In other words, use `eventTime = context.window().getEnd()`.